examples/ivr_outbound_dialer.py

```diff
@@ -0,0 +1,359 @@
+
+
+# Make an outbound call
+# This function will be called by originator for each call
+# User is supposed to provide the outbound DID or SIP URL for each call
+#
+# SIP Call
+#   dest_url='did@domain.com'   #Remote SIP URI
+#   dest_profile='internal'     #NSG defined SIP profile name
+#   dest_endpoint='sofia'       #For SIP calls one MUST set sofia
+#
+# FreeTDM Call
+#   dest_url='[A,a]/did'        #A=ascending hunt, a=descending hunt, DID number
+#   dest_profile='g1'           #profile is used as trunk group definition: g1 == group 1
+#   endpoint='freetdm"          #For TDM calls on MUST set freetdm
+#
+# In this example we are making a FreeTDM Call.
+# Change True to False in order to make SIP calls.
+def create_url():
+    if True:
+        # Make a FreeTDM SS7/PRI Call
+        # Adding F at the end of the DID disables remote SS7 overlap dialing which can add 5 sec to the incoming call setup time
+        # Note: Developer is suppose to supply their own DID. From a list or a DB
+        return {'dest_url': 'a/1000F', 'dest_profile': 'g1', 'dest_endpoint': 'freetdm'}
+    else:
+
+        # Make a SIP Call
+        # Uncoming in order to use it.
+        return {'dest_url': '4113@10.10.12.5:6060', 'dest_profile': 'internal', 'dest_endpoint': 'sofia'}
+
+
+# Create an originator
+# Originator is a dialer
+# You can tell it how many calls to make and at what frequency
+# After the first batch of calls are complete, you can choose to start dialing again.
+# There are 3 configurable variables
+#    max_calls_per_campaign
+#    max_call_attempts_per_sec
+#    max_campaigns
+# In my example the Dialer will dial out 2 calls as par to first campaign.
+# By increasing the max_campaigns, dialer will repeat as many dial campings.
+
+max_calls_per_campaign = 1
+max_call_attempts_per_sec = 1
+max_campaigns = 1
+
+originator = get_originator([(host, port)], apps=(IVRCallLogic,), auto_duration=False, rep_fields_func=create_url)
+
+# Use place holders in order for switch to trigger create_url() function above
+# TODO: Should be hidden in switchy
+#       Should iterate over all clients
+originator.pool.clients[0].set_orig_cmd(
+    dest_url='{dest_url}',
+    profile='{dest_profile}',
+    endpoint='{dest_endpoint}',
+    app_name='park',
+)
+
+
+# Setup calls per sec
+originator.rate = max_call_attempts_per_sec
+
+# Setup maximum number of calls to make
+originator.limit = max_calls_per_campaign
+
+# Maximum number of calls to dial out
+originator.max_offered = max_calls_per_campaign
+
+# Start the initial campaign
+# Originator will start making outbound calls
+originator.start()
+
+# Keeps a count of campaigns
+campaign_cnt = 0
+
+# Example of how to keep an eye on the campaign
+# After the campaign is over, check to see if another
+# campaign should start
+while (True):
+
+    log.info("Originator Stopped='{}' State='{}' Call Count='{}'\n" .  format(originator.stopped(), originator.state, originator.count_calls()))
```

Comment:
The `__repr__` method on the originator should print this all nicely for you by default.
I would just do `log.info(originator)` which will you something like:

`<Originator: active-calls=0 state=INITIAL total-originated-sessions=0 rate=30 limit=1 max-offered=inf duration=0>`